src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,12 +8,16 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
+    public static final String EXAMPLE = "123, some street, #12-34, 231534";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
     private boolean isPrivate;
+    private int numAddressComponents;
```

Comment:
Interesting, you attempted to make the address more flexible by allowing optional components. :-)

However, note that the problem description should be taken as a "user contract", i.e. we should try to adhere strictly to what they specify.